### PR TITLE
src: plugins: subsystems: drm: show current resolution and refresh rate

### PR DESCRIPTION
**Pull request by viniciusfl**
Solves [issue 1097](https://github.com/kworkflow/kworkflow/issues/1097).

### CODE REVIEW

**Comment by viniciusfl** [comment]
@rodrigosiqueira @davidbtadokoro

File: src/plugins/subsystems/drm/drm.sh [hunk 1]
```diff
@@ -345,8 +347,32 @@ function get_supported_mode_per_connector()
...
+  # get current resolution and hz
+  regex='mode: "[0-9]+x[0-9]+": [0-9]+'
+
   say 'Modes per card'
-  printf '%s\n' "$modes"
+  if [[ "$USER_ID" -eq 0 ]]; then
+    for file in "$SYSFS_KERNEL_DEBUG_DRI/"*"/state"; do
+      matches=$(grep -E "${regex}" "$file")
+      while IFS= read -r line; do
+        if [[ $line =~ $regex ]]; then
```
Comment (rodrigosiqueira) [inline src/plugins/subsystems/drm/drm.sh:358]: Add double-quotes around those bash variables.

File: src/plugins/subsystems/drm/drm.sh [hunk 2]
```diff
@@ -345,8 +347,32 @@ function get_supported_mode_per_connector()
...
+          resolution=$(sed -n 's/.*"\(.*\)".*/\1/p' <<< "${BASH_REMATCH[0]}")
```
Comment (rodrigosiqueira) [inline src/plugins/subsystems/drm/drm.sh:359]: Add a simple comment before your sed that describes (i.e., add an example) what kind of output we expected for `BASH_REMATCH`.

File: src/plugins/subsystems/drm/drm.sh [hunk 3]
```diff
@@ -345,8 +347,32 @@ function get_supported_mode_per_connector()
...
+          hz=$(awk -F': ' '{print $3}' <<< "${BASH_REMATCH[0]}")
+          break
+        fi
+      done <<< "$matches"
+    done
+
+    for line in $modes; do
+      if [[ $line == "$resolution" ]]; then
```
Comment (rodrigosiqueira) [inline src/plugins/subsystems/drm/drm.sh:367]: double quotes around $line

File: src/plugins/subsystems/drm/drm.sh [hunk 4]
```diff
@@ -345,8 +347,32 @@ function get_supported_mode_per_connector()
...
+
   say 'Modes per card'
-  printf '%s\n' "$modes"
+  if [[ "$USER_ID" -eq 0 ]]; then
+    for file in "$SYSFS_KERNEL_DEBUG_DRI/"*"/state"; do
+      matches=$(grep -E "${regex}" "$file")
+      while IFS= read -r line; do
+        if [[ $line =~ $regex ]]; then
+          resolution=$(sed -n 's/.*"\(.*\)".*/\1/p' <<< "${BASH_REMATCH[0]}")
+          hz=$(awk -F': ' '{print $3}' <<< "${BASH_REMATCH[0]}")
```
Comment (rodrigosiqueira) [inline src/plugins/subsystems/drm/drm.sh:360]: by hz, I think you mean refresh_rate.

File: tests/unit/drm_plugin_test.sh [hunk 5]
```diff
@@ -97,6 +101,33 @@ END
 END
 }
 
+function mk_fake_sys_kernel_debug_dri()
```
Comment (rodrigosiqueira) [inline tests/unit/drm_plugin_test.sh:104]: This function can be useful in other tests; move it to tests/unit/utils.sh.

File: tests/unit/drm_plugin_test.sh [hunk 6]
```diff
@@ -97,6 +101,33 @@ END
...
+{
+  declare -a fake_dirs=(
+    "1"
+    "128")
```
Comment (rodrigosiqueira) [inline tests/unit/drm_plugin_test.sh:108]: use single quotes.

File: tests/unit/drm_plugin_test.sh [hunk 7]
```diff
@@ -97,6 +101,33 @@ END
...
+
+  for dir in "${fake_dirs[@]}"; do
+    mkdir -p "${FAKE_DRI_SYSFS}/${dir}"
```
Comment (rodrigosiqueira) [inline tests/unit/drm_plugin_test.sh:111]: avoid short options.

File: tests/unit/drm_plugin_test.sh [hunk 8]
```diff
@@ -332,6 +363,39 @@ function test_get_supported_mode_per_connector()
   export SYSFS_CLASS_DRM="$FAKE_DRM_SYSFS"
   output=$(get_supported_mode_per_connector 2)
   compare_command_sequence '' "$LINENO" 'expected_output' "$output"
+
+  declare -a expected_output=(
```
Comment (rodrigosiqueira) [inline tests/unit/drm_plugin_test.sh:367]: Replace all double quotes with single quotes where we don't have a bash variable.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi,
Thanks for this PR; I added some comments. Also, your feature does not work as expected for remote machines or even for local machines since it is expected to be executed as root, which is not correct.

**Comment by cmhpedro** [comment]
> Hi, Thanks for this PR; I added some comments. Also, your feature does not work as expected for remote machines or even for local machines since it is expected to be executed as root, which is not correct.

Hi! Thanks for the response. We are accessing a kernel file, that needs `sudo` permission. The expected way is executing `kw ...` and requesting `sudo` permission (similar to `setup`)? If this is not the way, is there a suggestion on how to do it?

**Comment by rodrigosiqueira** [comment]
> > Hi, Thanks for this PR; I added some comments. Also, your feature does not work as expected for remote machines or even for local machines since it is expected to be executed as root, which is not correct.
> 
> Hi! Thanks for the response. We are accessing a kernel file, that needs `sudo` permission. The expected way is executing `kw ...` and requesting `sudo` permission (similar to `setup`)? If this is not the way, is there a suggestion on how to do it?

You don't need sudo to deploy to remote since it runs as root.

Also, are there any updates to my other comments?

**Comment by rodrigosiqueira** [comment]
Closed due to the lack of activity.

**Pull request closed without merging**